src/Enum.php

```diff
@@ -34,8 +28,7 @@ class Enum implements JsonSerializable
      */
     public function __construct($value)
     {
-        $this->reflection = new ReflectionClass(get_called_class());
-        $this->constants = $this->reflection->getConstants();
+        $this->constants = (new ReflectionClass(get_called_class()))->getConstants();
```

Comment:
@pies Its not using memory :) I found nice article about php reflection (but ofc I didn't save it). Seems to be - there is no different if you will cache reflection or not - it cached on engine level either way. And also - there is no difference how much reflection you want to use - how many objects etc. differences between thousands of objects and 1 object is few microseconds. Its why I don't have problem with that change :).